### ISSUE TITLE
Shares information remains in webUI even after `sharee` is deleted

### ISSUE THREAD

**Issue by SagarGi** [owncloud/ocis#8365]
## Describe the bug
When a `sharee` is deleted after some resource is shared to him/her. The share information is still visible for that user in the `Shares` panel to the `Sharer`.

## Steps to reproduce
1. As an `admin` create a user `test`
2. Create file `hello.txt`
3. Share `hello.txt` to user `test`
4. Delete user `test`
5. Navigate to `Shares` panel for file `hello.txt`

## Expected behavior
There should not have been information shared to user `test`

## Actual behavior
Since the user is already deleted but its information is still in the shares panel which should not have been.
 
![sharedDeletedUser](https://github.com/owncloud/web/assets/46086950/5109f464-94b5-4150-8ef4-fba776098b1a)

### Strange Behavior
After some times may be around a minute (do not know the exact time) or so the `sharee` information is deleted but the shares logo is still there in the UI as if the file is shared to some user but the shares list is empty. 

![shareLogo](https://github.com/owncloud/web/assets/46086950/3fe0181f-a578-4186-a464-bd0ed48142ba)

## Setup
Please describe how you started the server and provide a list of relevant environment variables or configuration files.

<details>
<p>

```console
OCIS= Infinite Scale 5.1.0-prealpha+4d5490a601
WEB= ownCloud Web UI 8.0.0-rc.2
```

</p>
</details>

**Comment by micbar** [comment]
the userinfo cache in ocis prevents that IMHO, but after some time, it should disappear.

The behavior like you describe is kind of expected.

**Comment by JanAckermann** [comment]
Could we maybe invalidate the cached shares, where the deleted user was part of, or to expensive ?
Maybe not for ocs but sharing ng. 

From my view I also don't see a problem but a regular user could be worried that the deleted user will still have access to the file (which isn't the case).

**Comment by saw-jan** [comment]
The deleted sharee gets disappeared from the collaborator list eventually. But the share indicator icon still remains for the resource. This is one of the issue in this report

![Screenshot from 2024-02-05 17-21-59](https://github.com/owncloud/ocis/assets/52366632/8129d7d7-f03c-40cd-b835-d319804cdf96)

CC @JanAckermann

**Comment by JanAckermann** [comment]
I think is is also cached ?

**Comment by JammingBen** [comment]
The indicator should be removed immediately if the last share has been removed and the resource has not been shared indirectly via a parent. We're updating the store directly in such case, so there shouldn't be any caching involved.

After a page reload, this information comes from the `shareTypes` property in the WebDAV request, so no OCS involvement.

Is the indicator still showing after a page reload, or is it gone then?

**Comment by saw-jan** [comment]
>Is the indicator still showing after a page reload, or is it gone then?

Yeah, the indicator remains after page reload. I waited for minutes after collaborator was not displayed in the list and still the indicator was there

**Comment by JammingBen** [comment]
Okay, yeah, then the WebDAV response seems to use that cached information as well. Weird though that the indicator still exists after the share is gone from the list.

**Comment by micbar** [comment]
> Okay, yeah, then the WebDAV response seems to use that cached information as well. Weird though that the indicator still exists after the share is gone from the list.

The indicator is still there because the file is still shared even if the user doesn't exist anymore.

Needs checking.

Currently we have no "post-delete" workflows because the "delete" is hard to track.

**Comment by rhafer** [comment]
> Could we maybe invalidate the cached shares where the deleted user was part of, or too expensive ?
> Maybe not for ocs but sharing ng.

Speaking for sharing-ng. We're not caching the received shares of users. Currently the received shares of a user are not even deleted if that user is being deleted. We are just not listing shares anymore, where the receiver can't be looked up.  We are however caching the results of userid to username lookups for some time (1min by default) using an in-memory cache.

That cache is NOT invalidated when a user is being deleted. Mainly because we can't do that reliably, especially in distributed setups, or when users are coming from some external system. See also: #8356
